### PR TITLE
Backport CPython extension (pyuwsgi) build improvements from master to 2.0

### PR DESCRIPTION
**Pull request by natefoo**
This change backports both the addition of `new_argc`/`new_argv` and the extensive improvements to the pyuwsgi plugin in #639 to 2.0. Because Linux binary wheels can now be uploaded to PyPI thanks to [manylinux](https://github.com/pypa/manylinux), this will allow for compiler-less `pip install uwsgi`.

### CODE REVIEW

File: uwsgi.h [hunk 1]
```diff
@@ -2785,6 +2785,10 @@ struct uwsgi_server {
 	struct uwsgi_string_list *touch_spoolers_reload;
 	int spooler_reload_mercy;
 
+	// uWSGI 2.1 backport
```
Comment (unbit) [inline uwsgi.h]: please move them at the end of the structure to maintain ABI compatibility

**Comment by xrmx** [comment]
@natefoo could you please squash it to bbbf899 and force push?

**Comment by natefoo** [comment]
@xrmx done, thanks!

**Comment by xrmx** [comment]
I meant just the struct fields commit, i can squash all the commits from github gui , it's fine anyway :)

**Comment by natefoo** [comment]
Ahh, yeah, I understand now. I redid the commits.

**Comment by natefoo** [comment]
Does this need anything else or is it looking good?

**Comment by xrmx** [comment]
Thanks for the patch, merged.

**Pull request merged**